stablehlo/transforms/Passes.h

```diff
@@ -61,6 +61,10 @@ void populateVhloToVersionPatterns(RewritePatternSet *patterns,
 void populateChloToStablehloPatterns(MLIRContext *context,
                                      RewritePatternSet *patterns);
 
+/// Collection of folding patterns for StableHLO.
+void populateStablehloAggressiveFolderPatterns(RewritePatternSet *patterns,
```

Comment:
Let's also make a `populateStablehloShapeFolderPatterns` which will have the subset needed by `StablehloRefineShapes`. That way when we add more patterns to this folder pass, they won't impact shape refinement

Reply:
I suppose the alternative would be to have shape refinement only classify certain ops as illegal so only those patterns apply.. I'm also OK with that approach, I'm happy to make that change in a followup.

Reply:
I like `populateStablehloShapeFolderPatterns` approach. In the next patch we can add `fold_float` option and pass it into convert pattern, but keep it constantly false for shape folders.

Reply:
I plan to add a new `tests/stablehlo_aggressive_folder.mlir` in the upcoming patches for iota and convert on floats.